backend/src/main/java/gov/cdc/usds/simplereport/service/TestOrderService.java

```diff
@@ -272,37 +345,48 @@ public AddTestResultResponse addTestResult(
               : new TestEvent(order, order.getCorrectionStatus(), order.getReasonForCorrection());
 
       TestEvent savedEvent = _terepo.save(testEvent);
-      order.setTestEventRef(savedEvent);
-      TestOrder savedOrder = _repo.save(order);
-
       resultEntity.setTestEvent(savedEvent);
       _resultRepo.save(resultEntity);
+      order.setTestEventRef(savedEvent);
+      TestOrder savedOrder = _repo.save(order);
       _testEventReportingService.report(savedEvent);
 
-      ArrayList<Boolean> deliveryStatuses = new ArrayList<>();
-
-      PatientLink patientLink = _pls.createPatientLink(savedOrder.getInternalId());
-      if (patientHasDeliveryPreference(savedOrder)) {
-
-        if (smsDeliveryPreference(savedOrder) || smsAndEmailDeliveryPreference(savedOrder)) {
-          boolean smsDeliveryStatus = testResultsDeliveryService.smsTestResults(patientLink);
-          deliveryStatuses.add(smsDeliveryStatus);
-        }
-
-        if (emailDeliveryPreference(savedOrder) || smsAndEmailDeliveryPreference(savedOrder)) {
-          boolean emailDeliveryStatus = testResultsDeliveryService.emailTestResults(patientLink);
-          deliveryStatuses.add(emailDeliveryStatus);
-        }
-      }
-
-      boolean deliveryStatus =
-          deliveryStatuses.isEmpty() || deliveryStatuses.stream().anyMatch(status -> status);
-      return new AddTestResultResponse(savedOrder, deliveryStatus);
+      return savedOrder;
     } finally {
       unlockOrder(order.getInternalId());
     }
   }
 
+  private void editMultiplexResults(TestOrder order, List<MultiplexTestResult> newResults) {
+    Set<Result> resultsToUpdate = order.getResultSet();
+    resultsToUpdate.forEach(
+        result -> {
+          SupportedDisease disease = result.getDisease();
+          newResults.stream()
+              .filter(r -> _diseaseService.getDiseaseByName(r.getDiseaseName()).equals(disease))
+              .findFirst()
+              .ifPresent(newResult -> result.setResult(newResult.getTestResult()));
+        });
+    _resultRepo.saveAll(resultsToUpdate);
+  }
+
+  private void saveMultiplexResults(
+      TestEvent event, TestOrder order, List<MultiplexTestResult> results) {
+    List<Result> resultsToSave =
+        results.stream()
+            .map(
+                result ->
+                    new Result(
+                        event,
+                        order,
+                        _diseaseService.getDiseaseByName(result.getDiseaseName()),
+                        result.getTestResult()))
+            .collect(Collectors.toList());
+    // Covid result already saved in saveTestResultToDatabase
+    resultsToSave.removeIf(result -> result.getDisease() == _diseaseService.covid());
```

Comment:
I think you could also add a `filter` to the stream to avoid creating the duplicate `Results` for covid.

Reply:
Yeah, that would probably make more sense 😅